src/algo/uint_tuple.h

```diff
@@ -56,22 +57,29 @@ constexpr size_t get_offset() {
   return _uint_tuple::accumulate(arr.begin(), arr.begin() + idx, size_t{0});
 }
 
-constexpr bool is_power_of_2(size_t x) { return !(x & (x - 1)); }
+template <typename Result, size_t size>
+constexpr Result generate_mask() {
+  if constexpr (sizeof(Result) * CHAR_BIT <= size) {
+    return -1;
```

Comment:
Will this cast OK?

What about uint64 and -1 (if -1 is 32 bit?)

Reply:
I'm sure that this will work as expected but lets just use numeric_limits to be sure.